File: solutions/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        int i = -1;
        int wordPos = 0;
        do {
            ++i;
            ++wordPos;
            if(sentence.substr(i, searchWord.size()) == searchWord) {
                return wordPos;
            }
        } while((i = sentence.find(' ', i)) != string::npos);
        return -1;
    }
};

File: solutions/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        j=0
        flag=False
        tmp=1
        res=0
        l=len(searchWord)
        if sentence[j:j+l]==searchWord:
            flag=True
            return tmp
        for i in range(len(sentence)):
            if sentence[i:i+l]==searchWord and sentence[i-1]==" ":
                flag=True
                res = i
                break
        
        for i in range(len(sentence)):
            if sentence[i]==" ":
                tmp+=1
            if i==res:
                break
        if flag:
            return tmp
        else:
            return -1